libs/wingsdk/src/target-sim/tokens.ts

```diff
@@ -61,8 +61,10 @@ export class SimTokens extends Tokens {
    * Binds the given token to the host.
    */
   public bindValue(host: IInflightHost, value: any) {
-    if (!(host instanceof Function)) {
```

Comment:
I think it's ok to remove this check now since we recently added the capability in the SDK that all IInflightHost are expected to implement addEnvironment - which is all the tokenizer needs.

Reply:
Over here:
```js
        const envName = this.envName(value);
        // the same token might be bound multiple times by different variables/inflight contexts
        if (host.env[envName] === undefined) {
          host.addEnvironment(envName, value);
        }
```

What if we change `addEnvironment` so you can call it multiple times with the same key, as long as the value that is already stored is the same? I'm not sure but I wonder if this could solve the use case

```js
host.addEnvironment("foo", "bar");
host.addEnvironment("foo", "bar"); // OK
host.addEnvironment("foo", "baz"); // runtime error!
```

Reply:
https://github.com/winglang/wing/pull/4379